UndertaleModLib/Compiler/Compiler.cs

```diff
@@ -106,7 +106,25 @@ public void Setup(bool redoAssets = false)
 
         private void MakeAssetDictionary()
         {
+            // Clear the dictionary first and set the worst case max size so that we don't resize it over and over
             assetIds.Clear();
+            int maxSize = 0;
```

Comment:
I think there should be a `Data` null check, so if it's null, then this method should only clear `assetIds` and `scripts`.
This will allow us to change `Data?.` to `Data.`.

Reply:
Nevermind, I am blind, I've already wrote that above...